Source/missiles.cpp

```diff
@@ -3034,13 +3029,13 @@ void MI_Firebolt(int i)
 			switch (missile._mitype) {
 			case MIS_FIREBOLT:
 			case MIS_MAGMABALL:
-				AddMissile(missile.position.tile, { i, 0 }, missile._mimfnum, MIS_MISEXP, missile._micaster, missile._misource, 0, 0);
+				AddMissile(missile.position.tile, missile.position.start, missile._mimfnum, MIS_MISEXP, missile._micaster, missile._misource, 0, 0);
 				break;
 			case MIS_FLARE:
-				AddMissile(missile.position.tile, { i, 0 }, missile._mimfnum, MIS_MISEXP2, missile._micaster, missile._misource, 0, 0);
+				AddMissile(missile.position.tile, missile.position.start, missile._mimfnum, MIS_MISEXP2, missile._micaster, missile._misource, 0, 0);
 				break;
 			case MIS_ACID:
-				AddMissile(missile.position.tile, { i, 0 }, missile._mimfnum, MIS_MISEXP3, missile._micaster, missile._misource, 0, 0);
+				AddMissile(missile.position.tile, missile.position.start, missile._mimfnum, MIS_MISEXP3, missile._micaster, missile._misource, 0, 0);

```

Comment:
these are hacks that are encoding the parent missile in the destination variable, are you sure you can just remove them?

Reply:
@AJenbo I tested firebolt only and firebolt seems to work fine. I will raise my Sorcerer and search for those books to see if other spells work well in the game :smile: 

![Screenshot from 2021-08-24 07-49-45](https://user-images.githubusercontent.com/6230110/130638707-cdf6d0a6-2185-4185-af77-c46af8254696.png)

Reply:
use the debug commands (chat):

`give xp` -> level up
`set spells 20` -> max level all spells
`god` -> invulnerable, no mana cost to spells

There are a lot of special condition and most of the missiles here are only cast by monsters in the game.

To check things like that it's bets to record a game play demo and then replay it after your changes to see if the change affect game play in a subtle way.

If you look in `AddMisexp()` you can see that your change here would actually cause random positioning of missiles explosions since it was abusing `dst.x` to mean the parent spell id. I think the best way to see this is to fight the eye-orbs in the hive.

Reply:
> @AJenbo I tested firebolt only and firebolt seems to work fine. I will raise my Sorcerer and search for those books to see if other spells work well in the game 😄
> 

You don't need to do anything manually anymore, just do
"dropitem book of x" with x being spell name

Reply:
@AJenbo and @qndel , thank you so much for the debug commands! You both saved me countless hours searching in the dungeons for books :laughing:

Reply:
> @AJenbo and @qndel , thank you so much for the debug commands! You both saved me countless hours searching in the dungeons for books 😆

I feel you ;) I'm so frustrated when I have to bisect older builds that don't have them :D

Reply:
You can also do `help` to see a list of commands, and `help <command>` to get extended help for a specific command.

Reply:
Ended up also fixing this one.